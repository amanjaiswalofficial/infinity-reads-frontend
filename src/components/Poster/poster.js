// Library imports
import React from 'react'

// Custom imports
import './poster.css'


const Poster = ({imageSrc}) => {


    return (
        <div style={{position: "relative"}}>
            <img data-testid="component-poster" src={imageSrc} alt="" className="wallpaper"/>
        </div>
    )
}

export default Poster
