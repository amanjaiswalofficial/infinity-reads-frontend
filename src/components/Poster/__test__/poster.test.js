// Library imports
import React from "react"
import ReactDOM from "react-dom"
import { render,  cleanup } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer"

// Custom imports
import Poster from 'components/Poster/poster'
import image from 'assets/img/wallpaper.png'

// Ensure each test case runs separately and cleans after it runs
afterEach(cleanup)

// Test case to ensure component created
it("renders without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(<Poster imageSrc={image}/>, div)
})

// Test case to ensure image loaded successfully
it("renders with correct image", () => {
    const { getByTestId } = render(<Poster imageSrc={image}/>)
    expect(getByTestId("component-poster")).toHaveAttribute("src", image)
})

// Test case to create and compare snapshot of poster component
// NOTE: Since we aren't applying any props or property to the component
// Hence no change can be tested, and kept only as reference for future
it("matches snapshot", () => {
    const tree = renderer.create(<Poster imageSrc={image}/>).toJSON()
    expect(tree).toMatchSnapshot()
})
