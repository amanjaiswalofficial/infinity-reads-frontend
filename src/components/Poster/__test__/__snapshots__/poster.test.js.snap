// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <img
    alt=""
    className="wallpaper"
    data-testid="component-poster"
    src="wallpaper.png"
  />
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "background": "#F67280",
        "border": "1px solid #F67280",
        "color": "#FFFFFF",
        "left": "70%",
        "position": "absolute",
        "top": "50%",
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Go To Site
    </span>
  </button>
  <a
    href="https://github.com/amanjaiswalofficial/infinity-reads-frontend/"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "border": "1px solid #F67280",
          "color": "#FFFFFF",
          "left": "79%",
          "position": "absolute",
          "top": "50%",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Github
      </span>
    </button>
  </a>
</div>
`;
